scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/edf/EDFPolicyExtended.java

```diff
@@ -160,4 +166,25 @@ private static Date getEffectiveExpectedExecutionTime(InternalJob internalJob, D
         }
     }
 
+    private void fireEventsIfSomeJobsWillNotMeetTheirDeadlines(List<JobDescriptor> jobs) {
+        Date now = new Date();
+        final List<InternalJob> jobsWhichWillMissDeadlines = jobs.stream()
+                                                                 .map(jobDescriptor -> ((JobDescriptorImpl) jobDescriptor).getInternal())
+                                                                 .filter(job -> getEffectiveDeadline(job,
+                                                                                                     now).compareTo(getEffectiveExpectedExecutionTime(job,
+                                                                                                                                                      now)) < 0)
+                                                                 .collect(Collectors.toList());
+
+        jobsWhichWillMissDeadlines.forEach(job -> {
+            LOGGER.warn(String.format("Job[id=%s] might miss its deadline (expected finish: %s after deadline: %s)",
+                                      job.getId().value(),
+                                      getEffectiveExpectedExecutionTime(job, now),
+                                      getEffectiveDeadline(job, now)));
+            try {
+                new JobDeadlineEmailNotification(job).doSend();
+            } catch (JobEmailNotificationException e) {
+                LOGGER.error(e);
```

Comment:
please use LOGGER.error("message", e) 
otherwise only the error message will be printed.